### ISSUE TITLE
Generic error when modifying database schema

### ISSUE THREAD

**Issue by dvekeman** [hasura/graphql-engine#3314]
I have a Postgresql database on Amazon RDS and Hasura running on docker (latest version v1.0.0-beta.9).

After applying a - rather intrusive - change to one of my tables(*) I can no longer make any modifications.

When I hit `'Data' > <table>` I can still query, insert, change permissions but when I hit the 'Modify' tab I get 

```
Error

Something went wrong. Head back Home.

keys@[native code]
n@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/main.js.gz:1:3386808
n@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/main.js.gz:1:629172
Ha@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2639864
fs@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2665502
ds@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2665892
Zs@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2673068
Ys@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2672436
nl@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2673824
Pr@https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2604215
Pr@[native code]
```

(*) the change I made - through the Hasura console - was not a regular change but I altered the primary key.
Originally the pk was an integer in an 'id' column. I added a new 'reference' column with a UUID, changed it to become the pk and then removed the original id column.
This 'person' table was also the table I used for row select filtering (`person.external_id == X-Hasura-User-Id`). So probably this was a bit too much and something broke. I tried to enable more logging but I guess it's something in the internal logs...?

**Comment by macs91** [comment]
I'm experiencing the same issue, when I hit "Modify" I get this error page:
```
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at n (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/main.js.gz:1:3386804)
    at Editor.n (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/main.js.gz:1:629171)
    at Ha (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2639858)
    at Wa (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2639651)
    at Ya (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2643090)
    at fs (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2665500)
    at ds (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2665890)
    at Zs (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2673066)
    at Ys (https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/vendor.js.gz:1:2672434)
```

**Comment by dvekeman** [comment]
I just restored my database to an earlier version and I am still seeing the issue. Would it be possible that it's unrelated to my database and rather caused by a change in the assets at https://graphql-engine-cdn.hasura.io/console/assets/channel/beta/v1.0/main.js?

**Comment by sawell121** [comment]
FYI setting the assets to load locally on Hasura rather than from the CDN temporarily fixed this issue for me. You will need to be on version v1.0.0-beta.1 or newer.

As per: https://docs.hasura.io/1.0/graphql/manual/deployment/graphql-engine-flags/reference.html

I set an environment variable on our docker-compose script to:

`HASURA_GRAPHQL_CONSOLE_ASSETS_DIR: /srv/console-assets`

Alternative method using the command flag:

`--console-assets-dir /srv/console-assets`

**Comment by rikinsk** [comment]
closed this in 3a07b2b
